pkg/healthcheck/healthcheck.go

```diff
@@ -600,6 +600,14 @@ func (hc *HealthChecker) allCategories() []category {
 						return hc.checkPodSecurityPolicies(true)
 					},
 				},
+				{
+					description: "pod injection disabled on kube-system (HA only)",
```

Comment:
Does it make sense to introduce a HA section? There'd only be one check there right now, so it might be a little weird.

Reply:
I toyed with this idea and reached the same conclusion. We can do it, if you think it makes sense though.

Reply:
could we programmatically add this check to the list of checks only when HA is enabled?  or is the config not yet available at that point?

Reply:
Yes, unfortunately the config is not there at that point.